### ISSUE TITLE
Please add Telegram notifications

### ISSUE THREAD

**Issue by shmelkin** [Xeio/WanderLost#75]
Hi Joshua!

I really like your website and me and my brother we use it every day. But not like it is intended. As we do not sit in our browsers all day, it is hard to follow whether an interesting piece of loot spawned. I programmed a crawler that checks ur website and notifies us via Telegram message everytime something desirable spawned. Unfortunately, everytime you edit your page layout, I have to reconfigure some of my script.
I think it would be a benefit if you added the option to register a Telegram account (or simply the user ID) in the configuration of your website to automatically send Telegram notifications via a bot. If needed and wanted, I can also assist you with that.

Best regards!

**Comment by Xeio** [comment]
So probably will not do Telegram specifically. There is #61 where I may add browser push notifications at some point.

As to crawling the site, you probably don't want to do that via a crawl anyway. It would generally be simpler to get a [SignalR](https://dotnet.microsoft.com/en-us/apps/aspnet/signalr) client library (should be available in most common languages) and connect directly to the https://lostmerchants.com/MerchantHub endpoint. The call interfaces are available [here](https://github.com/Xeio/WanderLost/tree/main/WanderLost/WanderLost/Shared/Interfaces), though it's mostly just calling the "SubscribeToServer" method with your server name, then listening to the "UpdateMerchant" mesasges from the server. If you just want to connect and get merchants at the time of connection, you can call "GetKnownActiveMerchantGroups" for the server though your client won't get real-time updates like that.

**Comment by shmelkin** [comment]
Thank you for responding and providing details.

I created a script that calls [GetKnownActiveMerchantGroups](https://github.com/Xeio/WanderLost/blob/9fe5979a34535d32b075bf8be58bc3a083cffb59/WanderLost/WanderLost/Shared/Interfaces/IMerchantHubServer.cs#L10) every minute to reduce load, as I don't really need real-time data.

Everything works as expected.

Is it possible to add expiring information and region into the activeMerchants object? Just so my notifications can include that information. Currently only zone is included, which seems incomplete.

**Comment by Xeio** [comment]
Expiration is always :55 after the hour. Things like region are in the data file https://github.com/Xeio/WanderLost/blob/main/WanderLost/WanderLost/Client/wwwroot/data/merchants.json

I don't push most of that to keep the message sizes smaller since the client can compute them.

**Comment by matiasa18** [comment]
@shmelkin can you share how you made the request to get that response? I'm interested in doing the same through NodeJS

**Comment by stefan0uh** [comment]
@matiasa18 in short
```js
import signalR from "@microsoft/signalr";

let connection = new signalR.HubConnectionBuilder()
  .withUrl("https://lostmerchants.com/MerchantHub")
  .build();

await connection.start();
await connection.invoke("SubscribeToServer", server);
await connection.on("UpdateMerchantGroup", async (server) => {
  let merchants = await connection.invoke(
      "GetKnownActiveMerchantGroups",
      server
    )
 console.log(merchants)
});
```

**Comment by matiasa18** [comment]
> @matiasa18 in short
> 
> ```js
> import signalR from "@microsoft/signalr";
> 
> const connection = new signalR.HubConnectionBuilder()
>   .withUrl("https://lostmerchants.com/MerchantHub")
>   .build();
> 
> await connection.start();
> await connection.invoke("SubscribeToServer", "Asta");
> await connection.on("UpdateMerchantGroup", async (server) => {
>   const merchants = await connection.invoke(
>       "GetKnownActiveMerchantGroups",
>       server
>     )
>  console.log(merchants)
> });
> ```

Thanks for this!
